src/InstallWizard.js

```diff
@@ -329,13 +329,26 @@ class InstallWizard extends Component {
   // Pages within the installer may request that the model be forceably loaded
   // from disk, espcially when a change is made to directly to the model files
   // to the model.  Returns a promise
-  loadModel = () => fetchJson('/api/v2/model')
-    .then(responseData => {
-      this.setState({'model': fromJS(responseData)});
-    })
-    .catch((error) => {
-      console.log('Unable to retrieve saved model');// eslint-disable-line no-console
-    })
+  loadModel = () => {
+    // If there is a pending promise to fetch the model return that promise.
+    if(!this.fetchModelPromise || this.fetchModelPromise?.done) {
```

Comment:
This still looks like it will run another fetch if the first one is still pending. I think it would be better to just use:
```
if (!this.fetchModelPromise) {
   this.fetchModelPromise = ... 
       (same stuff you have)
}
return this.fetchModelPromise;
```
***except*** remove assigning and checking for the `done` field

Reply:
With that logic this method will only be able to fetch the model once. Any subsequent calls will get the promise that was created the first time. What we could do, unset `fetchModelPromise` when it is resolved, instead of relying on the done flag.